packages/forms/frontend/sirius-components-widget-reference/src/modals/CreateModal.tsx

```diff
@@ -370,95 +371,90 @@ export const CreateModal = ({ editingContextId, widget, onClose, formId }: Creat
     }
   }, [createModal]);
 
-  const onBrowserSelection = (selection: Selection) => {
+  const onTreeItemClick = (_event, item: GQLTreeItem) => {
     const changeContainerSelectionEvent: ChangeContainerSelectionEvent = {
       type: 'CHANGE_CONTAINER_SELECTION',
-      container: selection,
+      container: { entries: [{ id: item.id, kind: item.kind }] },
```

Comment:
If I remember the code correctly, the other parts of the code are expecting a SelectionEntry, we could probably remove those references to the selection to use GQLTreeItem instead. I'll have to check.